packages/hub-extension/src/index.ts

```diff
@@ -56,11 +58,16 @@ function activateHubExtension(
 
   const { commands } = app;
 
-  // TODO: use /spawn/:user/:name
-  // but that requires jupyterhub 1.0
-  // and jupyterlab to pass username, servername to PageConfig
+  // requires jupyterhub 1.0
   const restartUrl =
-    hubHost + URLExt.join(hubPrefix, `spawn?next=${hubPrefix}home`);
+    hubHost +
```

Comment:
What if we switch here on whether there is a server name and choose the simplified url if there isn't one?  Does the simplified url work in jupyterhub 1.0?

Reply:
My understanding was that the default server name is the empty string (as @rcthomas said). If URLExt.join is smart and ignores empty strings as a no-op then presumably such a switch would be unnecessary. That being said I am not sure.

Reply:
See here: https://github.com/jupyterhub/jupyterhub/blob/master/jupyterhub/singleuser.py#L449

Reply:
The simplified URL should work in JupyterHub 1.0, but I think it's also synonymous with `/spawn/user` (i.e. `/spawn` and `/spawn/user` should be synonymous (assuming `'user'` corresponds to the person actually logged in). That being said you're right I'm not 100% sure and haven't looked at this in detail in a while.

I guess what I'm trying to say is that I think JupyterHub is written in a way that this can both be backwards-compatible and a one-liner, but again I need to test first, you're right.

Reply:
Nope sorry I think I was wrong: https://github.com/jupyterhub/jupyterhub/blob/1a26c1fb817f44acffd3907d0ec94d7de115c2e6/jupyterhub/handlers/pages.py#L131
You're right that switch is the right approach as far as I can tell.